pytensor/tensor/nlinalg.py

```diff
@@ -198,7 +198,15 @@ class Det(Op):
 
     def make_node(self, x):
         x = as_tensor_variable(x)
-        assert x.ndim == 2
+        if x.ndim != 2:
+            raise ValueError()
+        # Check for known shapes and square matrix
+        if all(shape is not None for shape in x.type.shape) and (
+            x.type.shape[0] != x.type.shape[1]
+        ):
```

Comment:
A bit more readable?
```suggestion
        if (None not in x.type.shape) and (x.type.shape[0] != x.type.shape[1]):
```